File: python/packages/aholo-sdk-core/src/manycore/aholo_sdk_core/ous_client.py
from __future__ import annotations

from .http import BaseHttpClient

OUS_TOKEN_HEADER = "ous-token-v2"


class OusHttpClient(BaseHttpClient):
    def __init__(self, *, base_url: str, ous_token: str, timeout_ms: int = 60_000) -> None:
        super().__init__(base_url=base_url, timeout_ms=timeout_ms, headers={OUS_TOKEN_HEADER: ous_token})
        self.ous_token = ous_token

File: typescript/packages/aholo-sdk-core/src/ous-client.ts
import { BaseHttpClient, type BaseHttpClientOptions } from './http.js';

const OUS_TOKEN_HEADER = 'ous-token-v2';

export interface OusClientOptions extends Omit<BaseHttpClientOptions, 'headers'> {
  ousToken: string;
}

/**
 * HTTP client for OUS `globalDomain` hosts (`ous-token-v2: <token>`).
 */
export class OusHttpClient extends BaseHttpClient {
  readonly ousToken: string;

  constructor(options: OusClientOptions) {
    super({
      ...options,
      baseUrl: options.baseUrl.replace(/\/$/, ''),
      headers: {
        [OUS_TOKEN_HEADER]: options.ousToken,
      },
    });
    this.ousToken = options.ousToken;
  }
}

export { OUS_TOKEN_HEADER };
